pkg/kubelet/kubelet.go

```diff
@@ -1500,7 +1500,18 @@ func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *api.ObjectFieldSelector, pod
 	}
 	switch internalFieldPath {
 	case "status.podIP":
-		return pod.Status.PodIP, nil
+		status, _ := kl.statusManager.GetPodStatus(pod.UID)
```

Comment:
Add some comments on ordering here @pmorie! We don't want a refactoring to screw up this logic later.